Source/Provers/SMTLib/SmtLibNameUtils.cs

```diff
@@ -64,54 +69,20 @@ public class SMTLibNamer : UniqueNamer
 
     static HashSet<string> reservedSmtWords;
     static bool[] validIdChar;
-    static bool symbolListsInitilized;
 
-    static void InitSymbolLists()
+    static SmtLibNameUtils()
     {
-      lock (reservedSmtWordsList)
+      reservedSmtWords = new HashSet<string>();
+      foreach (var w in reservedSmtWordsList)
       {
-        // don't move out, c.f. http://en.wikipedia.org/wiki/Double-checked_locking
-        if (symbolListsInitilized)
-        {
-          return;
-        }
-
-        reservedSmtWords = new HashSet<string>();
-        foreach (var w in reservedSmtWordsList)
-        {
-          reservedSmtWords.Add(w);
-        }
-
-        validIdChar = new bool[255];
-        for (int i = 0; i < validIdChar.Length; ++i)
-        {
-          validIdChar[i] = char.IsLetterOrDigit((char) i) || idCharacters.IndexOf((char) i) >= 0;
-        }
-
-        symbolListsInitilized = true;
+        reservedSmtWords.Add(w);
       }
-    }
 
-    static string AddQuotes(string s)
-    {
-      var allGood = true;
-
-      foreach (char ch in s)
+      validIdChar = new bool[255];
```

Comment:
Why 255? I'm guessing it has to do with 255 being close to 2^8. But then, why not use all 256? Or is 255 just an arbitrary "pretty small" number?

Reply:
I don't know, I didn't write this code, just moved it.